astropy/visualization/wcsaxes/tests/datasets.py

```diff
@@ -0,0 +1,62 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Downloads the FITS files that are used in image testing and for building documentation.
+"""
+
+import time
+
+from ....utils.data import download_file
+from ....io import fits
+
+__all__ = ['fetch_msx_hdu',
+           'fetch_rosat_hdu',
+           'fetch_twoMASS_k_hdu',
+           'fetch_l1448_co_hdu',
+           'fetch_bolocam_hdu',
+           ]
+
+MAX_RETRIES = 10
+TIME_BETWEEN_RETRIES = 5
+URL = 'http://data.astropy.org/'
+
+
+def fetch_hdu(filename, cache=True):
+    """Download a FITS file to the cache and open HDU 0.
+    """
+    for retry in range(MAX_RETRIES):
+        try:
+            path = download_file(URL + filename, cache=cache, timeout=30)
+        except:
+            if retry == MAX_RETRIES - 1:
+                raise
+            else:
+                time.sleep(TIME_BETWEEN_RETRIES)
+        else:
+            break
+    return fits.open(path)[0]
```

Comment:
Local variable `path` might be referenced before assignment.
I think that's acually right, it could result in a `NameError` instead of a good error message, no?